src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -165,111 +230,50 @@ public boolean publish(SlackRequest slackRequest) {
                         .collect(Collectors.toList())
                 );
             }
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "Error closing HttpClient", e);
+        }
 
-            for (String roomId : roomIds) {
-                HttpPost post;
-                String url;
-                String threadTs = "";
-
-                //thread_ts is passed once with roomId: Ex: roomId:threadTs
-                String[] splitThread = roomId.split("[:]+");
-                if (splitThread.length > 1) {
-                    roomId = splitThread[0];
-                    threadTs = splitThread[1];
-                }
-                JSONObject json = new JSONObject();
-                json.put("channel", roomId);
-                if (StringUtils.isNotEmpty(message)) {
-                    json.put("text", message);
-                }
-                if (attachments != null && !attachments.isEmpty()) {
-                    json.put("attachments", attachments);
-                }
-
-                if (blocks != null && !blocks.isEmpty()) {
-                    json.put("blocks", blocks);
-                }
-                json.put("link_names", "1");
-                json.put("unfurl_links", "true");
-                json.put("unfurl_media", "true");
+        for (String roomId : roomIds) {
```

Comment:
This code is dedented because we're not wrapping it in `try (CloseableHttpClient...`, but is otherwise very similar to what was here before.